domain/TenantAdminOrganization.go

```diff
@@ -9,17 +18,228 @@ package domain
 type (
 	TenantAdminOrganizationIn struct {
 		RequestCommon
+		Cmd         string       `json:"cmd" form:"cmd" query:"cmd" long:"cmd" msg:"cmd"`
+		User        rqAuth.Users `json:"user" form:"user" query:"user" long:"user" msg:"user"`
+		Id          uint64       `json:"id" form:"id" query:"id" long:"id" msg:"id"`
+		NewParentId uint64       `json:"newParentId" form:"newParentId" query:"newParentId" long:"newParentId" msg:"newParentId"`
+		OldParentId uint64       `json:"oldParentId" form:"oldParentId" query:"oldParentId" long:"oldParentId" msg:"oldParentId"`
 	}
+
 	TenantAdminOrganizationOut struct {
 		ResponseCommon
+		Orgs []rqAuth.Orgs
 	}
 )
 
 const (
 	TenantAdminOrganizationAction = `tenantAdmin/organization`
+
+	ErrTenantAdminOrganizationUnauthorized                 = `unauthorized user`
+	ErrTenantAdminOrganizationTenantNotFound               = `tenant admin not found`
+	ErrTenantAdminOrganizationOldParentNotFound            = `old parent of the organization cannot be found`
+	ErrTenantAdminOrganizationNewParentNotFound            = `new parent of the organization cannot be found`
+	ErrTenantAdminOrganizationUpdatedOldParentChildren     = `cannot update childs of old parent`
+	ErrTenantAdminOrganizationUpdatedNewParentChildren     = `cannot update childs of new parent`
+	ErrTenantAdminOrganizationUpdatedParentForOrganization = `cannot update parent for organization`
+	ErrTenantAdminOrganizationNotFound                     = `cannot find the organization`
+	ErrTenantAdminOrganizationChildsNotFound               = `cannot find childs for update`
+	ErrTenantAdminOrganizationUpdatedChilds                = `cannot update childs`
 )
 
 func (d *Domain) TenantAdminOrganization(in *TenantAdminOrganizationIn) (out TenantAdminOrganizationOut) {
 	defer d.InsertActionLog(&in.RequestCommon, &out.ResponseCommon)
+
+	sess := d.MustLogin(in.RequestCommon, &out.ResponseCommon)
+	if sess == nil {
+		return
+	}
+
+	user := rqAuth.NewUsers(d.AuthOltp)
+	user.Id = sess.UserId
+	if !user.FindById() {
+		out.SetError(fiber.StatusBadRequest, ErrTenantAdminOrganizationUnauthorized)
+		return
+	}
+
+	tenant := rqAuth.NewTenants(d.AuthOltp)
+	tenant.TenantCode = user.TenantCode
+	if !tenant.FindByTenantCode() {
+		out.SetError(400, ErrTenantAdminDashboardTenantNotFound)
+		return
+	}
+
+	org := rqAuth.NewOrgs(d.AuthOltp)
+	out.Orgs = org.FindOrgsByTenantCode(tenant.TenantCode)
+
+	switch in.Cmd {
+	case zCrud.CmdUpsert:
+		orgW := wcAuth.NewOrgsMutator(d.AuthOltp)
+		isFamilyWithNewParent := isFamily(in.Id, in.NewParentId, out.Orgs)
+		currentOrg := findLocalOrgById(out.Orgs, in.Id)
+
+		if in.NewParentId == in.OldParentId {
```

Comment:
kalau parent sama, hanya reorder children harusnya, perlu tambahkan `in.OrderPosition` buat tahu harus ditaruh posisi ke berapa di parent.Children